graphql/schemabuilder/pagination.go

```diff
@@ -514,22 +514,22 @@ func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interfa
 
 	g, ctx := errgroup.WithContext(ctx)
 	matchStrings := filter.GetMatchStrings(*args.FilterText)
-
-	expensiveNodesToKeep, err := c.applyTextFilterNotBatchedExpensive(ctx, nodes, matchStrings, filterTextFieldsNotBatchedExpensive)
-	if err != nil {
-		return nil, err
-	}
-
-	nodesToKeep, err := c.applyTextFilterNotBatched(ctx, nodes, matchStrings, filterTextFieldsNotBatched)
-	if err != nil {
-		return nil, err
-	}
-
-	batchedNodesToKeep, err := c.applyBatchTextFilter(ctx, nodes, matchStrings, filterTextFieldsBatched)
-	if err != nil {
-		return nil, err
-	}
-
+	nodesToKeep := make([]bool, len(nodes))
+	expensiveNodesToKeep := make([]bool, len(nodes))
+	batchedNodesToKeep := make([]bool, len(nodes))
```

Comment:
It looks like we're allocating memory for these slices twice, once here and once in the helper functions?  Probably we can just declare these:
```
var nodesToKeep []bool
var expensiveNodesToKeep []bool
var batchedNodesToKeep []bool
```

Reply:
hm thought this would work, but it expects that array to be a certain length and gets out of index errors otherwise. However, the maps are populated since the filtering tests work correctly.

Reply:
maybe I'm misunderstanding how Go slice assignment works... I would have thought that line 519 would copy the slice data to the `nodesToKeep` value, overwriting the pointer.  So the old memory would be garbage-collected, and the `nodesToKeep` value would point to the memory allocated on line 462.  Is that not the case?

Reply:
discussed and resolved offline